proof/crefine/RISCV64/IpcCancel_C.thy

```diff
@@ -2915,9 +2916,13 @@ sorry (* FIXME RT: reply_remove_tcb_corres *)
 
 lemma reply_unlink_ccorres:
   "ccorres dc xfdc
-    (invs' and tcb_at' tcbPtr and reply_at' replyPtr)
-    (\<lbrace>\<acute>tcb = tcb_ptr_to_ctcb_ptr tcbPtr\<rbrace> \<inter> \<lbrace>\<acute>reply = Ptr replyPtr\<rbrace>) []
-    (replyUnlink tcbPtr replyPtr) (Call reply_unlink_'proc)"
+    (valid_tcbs' and pspace_aligned' and pspace_distinct'
+       and st_tcb_at' (\<lambda>st. (\<exists>oref cg. st = Structures_H.BlockedOnReceive oref cg (Some replyPtr))
+                            \<or> st = Structures_H.BlockedOnReply (Some replyPtr))
+                      tcbPtr
+       and obj_at' (\<lambda>reply. replyTCB reply = Some tcbPtr) replyPtr)
```

Comment:
indentation: the advantage of putting operators on the left is we can align them with the previous line in most cases (see style guide), so `and` should align with `valid_tcbs'`.
Also, the `∨` belongs under the `∃`, i.e. under `st =`

Reply:
I've fixed the indentation level of the `and`s.

As for the `∨`, I put it under the `(` that precedes the `∃` because the structure of the term is `(∃ ... st = ...) ∨ st = ...`. Could you clarify where the  `∨` belongs and why?

Reply:
You put it in the right place, for exactly the reasons you stated. If you look at the `∨`, it should be to the right of whatever it belongs to, i.e. the `λst.`, and it is on the same level as the LHS argument of the `∨`, which is the `(∃ ... st = ...)`. This means that the structure can be seen at a glance, which is important when doing textual surgery on theory files (which is diffs, git conflicts, PRs, etc).